File: 01268.cpp
#include <iostream>
using namespace std;

int students[1001][6];
bool friends[1001][1001];

int main() {
	int n; cin >> n;
	for (int i = 1; i <= n; i++)
		for (int j = 1; j < 6; j++)
			cin >> students[i][j];

	for (int i = 1; i <= n; i++)
		for (int j = 1; j <= n; j++)
			friends[i][j] = false;

	for (int i = 1; i <= n; i++) {
		// i: 학생 수
		for (int j = 1; j < 6; j++) {
			// j: 학년
			for (int k = i + 1; k <= n; k++) {
				// k: i번째 학생 제외한 다른 친구들
				if (students[i][j] == students[k][j]) {
					friends[i][k] = friends[k][i] = true;
				}
			}
		}
	}

	int max_cnt = -1, master = 0;
	for (int i = 1; i <= n; i++) {
		int cnt = 0;
		for (int j = 1; j <= n; j++)
			if (friends[i][j]) cnt++;

		if (cnt > max_cnt) {
			max_cnt = cnt;
			master = i;
		}
	}

	cout << master << endl;
}

File: 01268.py
n = int(input())

students = []
friends = []
for i in range(n):
    students.append(list(map(int, input().split())))
    friends.append([False] * n)

for i in range(n):
    for j in range(5):
        for k in range(i + 1, n):
            if students[i][j] == students[k][j]:
                friends[i][k] = friends[k][i] = True

max_cnt = -1; master = 0
for i in range(n):
    cnt = 0
    for j in range(n):
        if friends[i][j]: cnt += 1

    if cnt > max_cnt:
        max_cnt = cnt
        master = i + 1

print(master)
